### PR TITLE
implement package_ensure param for archlinux

### PR DESCRIPTION
**Pull request by bastelfreak**
we can't pass the grafana version to the archlinux package provider. it
has no versionable feature. this adds a workaround to provide 'present'
to it. This can later be extended to pass explicit states to the other
operating systems as well, or to implement ensure=>absent. We can't set
the $version variable to 'present' because this will kill the other
installation methods.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -123,7 +123,7 @@
             fail('manage_package_repo is not supported on Archlinux')
           }
           package { $::grafana::package_name:
-            ensure  => $::grafana::version, # pacman provider doesn't have feature versionable
+            ensure  => $::grafana::package_ensure, # pacman provider doesn't have feature versionable
```
Comment (alexjfisher) [inline manifests/install.pp]: This looks problematic.  If a user has provided their own value for `version` it won't get used anymore?

Comment (bastelfreak) [inline manifests/install.pp]: this will never ever work on archlinux. the archlinux package manager doesn't understand versions.

Comment (alexjfisher) [inline manifests/install.pp]: `package_ensure` needs to default to `$::grafana::version` not `$::grafana::params::version`.

Comment (alexjfisher) [inline manifests/install.pp]: @bastelfreak Sure, but you can't break it for all other Linuxes meanwhile.

File: manifests/params.pp [hunk 2]
```diff
@@ -24,12 +24,14 @@
       $install_method      = 'repo'
       $cfg_location        = '/etc/grafana.ini'
       $service_name        = 'grafana'
+      $package_ensure      = 'present'
```
Comment (alexjfisher) [inline manifests/params.pp]: param docs said it defaults to `latest`, not `present`

File: manifests/install.pp (hunk 1 above)
Comment (alexjfisher) [inline manifests/install.pp]: Maybe use something like...
```puppet
$ensure = pick($::grafana::package_ensure, $::grafana::version)
package { $::grafana::package_name:
  ensure = $ensure,
...
```
then default your new `package_ensure` parameter to `undef` for OSes other than ArchLinux??

**Pull request merged**